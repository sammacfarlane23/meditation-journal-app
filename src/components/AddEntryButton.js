import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export default () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const dispatch = useDispatch();

  const showModal = () => {
    setIsModalOpen(true);
  };

  const onSubmit = (entry) => {
    dispatch(startCreateEntry(entry));
    setIsModalOpen(false);
  };

  const closeModal = () => {
    setIsModalOpen(false);
  };

  return (
    <div className='button-position'>
      {!isModalOpen && (
        <button className='button--big circular-container' onClick={showModal}>
          <FontAwesomeIcon icon={faPlus} size='2x' />
        </button>
      )}
      <EntryModal
        showModal={isModalOpen}
        onSubmit={onSubmit}
        closeModal={closeModal}
      />
    </div>
  );
};
