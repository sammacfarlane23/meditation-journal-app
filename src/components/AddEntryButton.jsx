import React from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export class AddEntryButton extends React.Component {
  state = { show: false };

  showModal = () => {
    this.setState(() => ({ show: true }));
  };

  onSubmit = (entry) => {
    this.props.startCreateEntry(entry);
    this.setState({ show: false });
  };

  closeModal = () => {
    this.setState({ show: false });
  };
  render() {
    return (
      <div className='button-position'>
        {!this.state.show && (
          <button
            className='button--big circular-container'
            onClick={this.showModal}
          >
            <FontAwesomeIcon icon={faPlus} size='2x' />
          </button>
        )}
        <EntryModal
          showModal={this.state.show}
          onSubmit={this.onSubmit}
          closeModal={this.closeModal}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateEntry: (entry) => dispatch(startCreateEntry(entry)),
});

export default connect(undefined, mapDispatchToProps)(AddEntryButton);
