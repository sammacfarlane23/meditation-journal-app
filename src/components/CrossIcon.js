import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

export default (props) => (
  <div>
    <button className='footer-button' onClick={props.showSearch}>
      <FontAwesomeIcon icon={faTimes} className='white-button' size='2x' />
    </button>
  </div>
);
