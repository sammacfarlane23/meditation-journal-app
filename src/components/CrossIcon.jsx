import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

export default (props) => (
  <div>
    <button className='button--big' onClick={props.showSearch}>
      <FontAwesomeIcon icon={faTimes} size='2x' />
    </button>
  </div>
);
