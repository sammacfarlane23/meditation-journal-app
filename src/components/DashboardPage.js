import React from 'react';
import EntryList from './EntryList';
import Footer from './Footer';

export default () => (
  <div className='dashboard-layout'>
    <div className='content-container'>
      <EntryList />
    </div>
    <Footer />
  </div>
);
