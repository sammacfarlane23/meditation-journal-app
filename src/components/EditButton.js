import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEdit } from '@fortawesome/free-solid-svg-icons';

export default () => (
  <div>
    <FontAwesomeIcon icon={faEdit} className='edit-icon' />
  </div>
);
