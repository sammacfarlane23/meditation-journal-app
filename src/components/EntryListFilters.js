import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { setTextFilter } from "../actions/filters";

const EntryListFilters = () => {
  const dispatch = useDispatch();
  const filters = useSelector((state) => state.filters);

  const onTextChange = (e) => {
    dispatch(setTextFilter(e.target.value));
  };

  return (
    <div>
      <input
        type="text"
        aria-label="search-input"
        autoFocus
        placeholder="Search entries"
        value={filters.text}
        onChange={onTextChange}
        className="search-bar"
      ></input>
    </div>
  );
};

export default EntryListFilters;
