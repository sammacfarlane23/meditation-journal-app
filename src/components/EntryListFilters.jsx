import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

export class EntryListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  render() {
    return (
      <div>
        <input
          type='text'
          autoFocus
          placeholder='Search entries'
          value={this.props.filters.text}
          onChange={this.onTextChange}
          className='search-bar'
        ></input>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EntryListFilters);
