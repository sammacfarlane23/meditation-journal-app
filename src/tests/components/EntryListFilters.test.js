import React from 'react';
import { shallow } from 'enzyme';
import { EntryListFilters } from '../../components/EntryListFilters';

test('should render EntryListFilters correctly', () => {
  const text = '';
  const wrapper = shallow(<EntryListFilters filters={{ text }} />);
  expect(wrapper).toMatchSnapshot();
});
