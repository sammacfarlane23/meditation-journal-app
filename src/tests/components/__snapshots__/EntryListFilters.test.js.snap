// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EntryListFilters correctly 1`] = `
<div>
  <input
    autoFocus={true}
    className="search-bar"
    onChange={[Function]}
    placeholder="Search entries"
    type="text"
    value=""
  />
</div>
`;
